apps/cloudbroker/base/cloudapi__machines/methodclass/cloudapi_machines.py

```diff
@@ -1788,24 +1788,28 @@ def updateUser(self, machineId, userId, accesstype, **kwargs):
         return self._updateACE(machineId, userId, accesstype, userstatus)
 
     @authenticator.auth(acl={"cloudspace": set("X")})
-    def attachExternalNetwork(self, machineId, **kwargs):
+    def attachExternalNetwork(self, machineId, externalNetworkId, **kwargs):
         """
          Attach a external network to the machine
 
         :param machineId: id of the machine
+        :param externalNetworkId: optional external network id to take ip address from
         :return: True if a external network was attached to the machine
         """
         provider, node, vmachine = self.cb.getProviderAndNode(machineId)
+        cloudspace = self.models.cloudspace.get(vmachine.cloudspaceId)
+        externalNetId = externalNetworkId or cloudspace.externalnetworkId
```

Comment:
if externalnetworkId is empty we probably want to search for the best available externalnetwork (see network.py module) Intead of taking the same as the cloudspace (external network might have been exhausted since)